app/components/Progress.tsx

```diff
@@ -1,50 +1,140 @@
-import * as React from "react"
-import { StyleProp, TextStyle, View, ViewStyle } from "react-native"
-import { observer } from "mobx-react-lite"
-import { spacing } from "app/theme"
-import { Text } from "app/components/Text"
-
-export interface ProgressProps {
-  /** a required prop to specify the progress. */
-  progress: number
-  /** an optional style override useful for padding & margin. */
-  style?: StyleProp<ViewStyle>
+// Packages Imports
```

Comment:
Please make a different component for circular progress and keep this component intact, alternatively, you can make a preset in this component itself. Whatever suits best to you.